python/splendor/human.py

```
@@ -0,0 +1,686 @@
+    """
+    key = raw.lower()
+
+    # full color names first
+    if key in BASE_COLORS:
+        return key  # type: ignore[return-value]
+
+    # abbreviations
+    if key in COLOR_ABBR_TO_FULL:
+        return COLOR_ABBR_TO_FULL[key]
+
+    raise ValueError(f"Unknown color: {raw}")
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)